docs/website/docs/reference/optimization-options.md

```diff
@@ -0,0 +1,54 @@
+# Optimization Options
+
+This page documents various supported flags for optimizing IREE programs. Each
+is presented with its English name, flag to enable/disable and default state.
+
+These flags can be passed to the:
+
+* `ireec` command line tool
+* `extra_args=["--flag"]` argument to `iree.compiler.tools` Python wrappers
+* In-process Python compiler API
+  `iree.compiler.transforms.ireec.CompilerOptions("--flag", "--flag2")`
+  constructor
+* `ireeCompilerOptionsSetFlags()` compiler C API function
+
+## High level program optimizations
+
+### Constant Evaluation (`--iree-opt-const-eval` (off))
+
+Performs compile-time evaluation of any global initializers which produce
+the initial values for global constants, storing the global directly in the
+program as constant data. This extracts such constant program fragments and
+recursively compiles them, using the runtime to evaluate the results.
+
+Note that this only has any effect on computations in module initializer
+functions, not free-standing operations in the program which may produce
+constant-derived results. See `--iree-opt-const-expr-hoisting` for options to
+optimize these.
```

Comment:
I feel like some of this could be considered implementation details. I'm not sure what percentage of the audience for this document would care _how_ this constant evaluation is performed (recursive compilation using the runtime to evaluate results) vs primarily caring about the costs/benefits of flipping this flag.

sample costs: longer compile times, higher peak memory usage
sample benefits: faster runtime for models with X behavior, lower peak memory usage

Reply:
It's an early feature that can have implications and should only be turned on if you understand it. I'll leave this level of detail for now. As a counter-argument: https://gcc.gnu.org/onlinedocs/gcc/Optimize-Options.html

Reply:
👍 thanks for the gcc link! I've been looking for something like that to compare docs against.